torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1051,6 +1052,74 @@ int listClear(Stack& stack) {
   return 0;
 }
 
+template <typename TList, typename TElement>
+int listInsert(Stack& stack) {
+  TList list;
+  int64_t idx;
+  TElement elem;
+  pop(stack, list, idx, elem);
+
+  auto& elements = list->elements();
+  const int64_t list_size = elements.size();
+  const int64_t normalized_idx = normalizeIndex(idx, list_size);
+
+  if (normalized_idx < 0 || normalized_idx >= list_size) {
+    if (normalized_idx < 0) {
+      elements.insert(elements.begin(), elem);
+    } else {
+      elements.push_back(elem);
+    }
+  } else {
+    elements.insert(elements.begin() + normalized_idx, elem);
+  }
+
+  return 0;
+}
+
+template <typename TList, typename TElement>
+int listRemove(Stack& stack) {
+  TList list;
+  TElement elem;
+  pop(stack, list, elem);
+
+  auto& elements = list->elements();
+  auto pos = std::find(elements.begin(), elements.end(), elem);
+
+  if (pos != elements.end()) {
+    elements.erase(pos);
+  } else {
+    AT_ERROR("list.remove(x): x not in list");
+  }
+
+  return 0;
+}
+
+template <>
+int listRemove<Shared<TensorList>, at::Tensor>(Stack& stack) {
+  Shared<TensorList> list;
+  at::Tensor elem;
+  pop(stack, list, elem);
+
+  auto& elements = list->elements();
+  auto pos = std::find_if(elements.begin(), elements.end(), [elem](const at::Tensor& b) {
+    const auto cmp_result = elem.eq(b);
+    return cmp_result.is_nonzero();
+  });
+
+  if (pos != elements.end()) {
+    elements.erase(pos);
+  } else {
+    AT_ERROR("list.remove(x): x not in list");
+  }
+
+  return 0;
+}
+
+template <>
+int listRemove<Shared<GenericList>, IValue>(Stack& stack) {
+  AT_ERROR("list.remove is not supported yet for 'GenericList'");
```

Comment:
Could you remove this, remove "aten::remove" from the MUTABLE_LIST_OPS macro, and then add add aten::remove with TensorList? 

That way when someone tries to use remove on a generic list the error will be printed out in schema matching, with the source location of the code.

Reply:
done!